frontend/pages/recipe/_slug/ocr-editor.vue

```diff
@@ -0,0 +1,924 @@
+      state.imagePosition.scale = state.imagePosition.dWidth / image.width;
+
+      // force the original ratio to be respected
+      state.imagePosition.dHeight = image.height * state.imagePosition.scale;
+
+      // Don't let images bigger than the canvas be zoomed in more than 1:1 scale
+      // Meaning only let images smaller than the canvas to have a scale > 1
+      if (!state.isImageSmallerThanCanvas && state.imagePosition.scale > 1) {
+        state.imagePosition.scale = 1;
+      }
+    }
+
+    /**
+     * This function will find the title of a recipe with the assumption that the title
+     * has the biggest ratio of surface area / number of words on the image.
+     * @return Returns the text parts of the block with the highest score.
+     */
+    function findRecipeTitle() {
+      const filtered = tsv.value.filter((element) => element.level === 2 || element.level === 5);
+      const blocks = [[]] as OcrTsvResponse[][];
+      let blockNum = 1;
+      filtered.forEach((element, index, array) => {
+        if (index !== 0 && array[index - 1].blockNum !== element.blockNum) {
+          blocks.push([]);
+          blockNum = element.blockNum;
+        }
+        blocks[blockNum - 1].push(element);
+      });
+
+      let bestScore = 0;
+      let bestBlock = blocks[0];
+      blocks.forEach((element) => {
+        // element[0] is the block declaration line containing the blocks total dimensions
+        // element.lenght is the number of words (+ 2) contained in that block
+        const elementScore = (element[0].height * element[0].width) / element.length; // Prettier is adding useless parenthesis for a mysterious reason
+        const elementText = element.map((element) => element.text).join(""); // Identify empty blocks and don't count them
+        if (elementScore > bestScore && elementText !== "") {
+          bestBlock = element;
+          bestScore = elementScore;
+        }
+      });
+
+      return bestBlock
+        .filter((element) => element.level === 5 && element.conf >= 40)
+        .map((element) => {
+          return element.text.trim();
+        })
+        .join(" ");
+    }
+
+    onMounted(() => {
+      invoke(async () => {
+        await until(recipe).not.toBeNull();
+
+        if (!recipe.value || !recipe.value.assets || !recipe.value.slug) {
+          return;
+        }
+
+        const assetName = recipe.value?.assets[0].fileName as string;
+        const imagesrc = assetURL(assetName);
+        image.src = imagesrc;
+
+        const res = await api.ocr.assetToTsv(recipe.value.slug, assetName);
+        tsv.value = res.data as OcrTsvResponse[];
+        state.loading = false;
+
+        if (recipe.value.name?.match(/New\sOCR\sRecipe(\s\([0-9]+\))?/g)) {
+          recipe.value.name = findRecipeTitle();
+        }
+
+        nextTick(() => {
+          state.canvas = <HTMLCanvasElement>document.getElementById("canvas");
+          state.ctx = <CanvasRenderingContext2D>state.canvas.getContext("2d");
```

Comment:
You should use [template refs](https://vuejs.org/guide/essentials/template-refs.html#template-refs) to access dom elements instead of using document.getElementById.